src/main.rs

```diff
@@ -42,15 +68,138 @@ async fn main() {
+                        }
+                    }
+                };
+                tokio::spawn(fut);
+            }
+
+            // subscribe `rejected_transaction` topic
+            {
+                let typed_client: TypedClient = rpc_channel.clone().into();
+                let pool = pool.clone();
+                let fut = async move {
+                    match typed_client.subscribe::<_, String>(
+                        "subscribe",
+                        ["rejected_transaction"],
+                        "subscribe",
+                        "unsubscribe",
+                        "String",
+                    ) {
+                        Ok(mut stream) => loop {
+                            if let Some(subscription) = stream.next().await {
+                                match subscription {
+                                    Ok(json_string) => {
+                                        debug!(
+                                            "Rpc subscription notified a rejected transaction: {}",
+                                            json_string
+                                        );
+                                        if let Ok((pool_tx_entry, _)) = serde_json::from_str::<(
+                                            PoolTransactionEntry,
+                                            PoolTransactionReject,
+                                        )>(
+                                            &json_string
+                                        ) {
+                                            let tx: Transaction =
+                                                pool_tx_entry.transaction.inner.into();
+                                            pool.write()
+                                                .expect("acquire lock")
+                                                .transaction_rejected(&tx.into_view());
+                                        } else {
+                                            error!("Failed to parse json_string: {}", json_string);
+                                        }
+                                    }
+                                    Err(err) => {
+                                        error!("Rpc subscription error {:?}", err);
+                                    }
+                                }
+                            }
+                        },
+                        Err(err) => {
+                            error!("subscribe error {:?}", err);
+                        }
+                    }
+                };
+                tokio::spawn(fut);
+            }
+        }
+
+        service.poll(rpc_channel.into()).await;
+    } else if !uri.starts_with("http") {
+        if index_tx_pool {
+            error!("indexing the pending txs in the ckb tx-pool is only supported when connecting to ckb rpc service with tcp protocol")
+        } else {
+            uri = format!("http://{}", uri);
```

Comment:
`if !uri.starts_with("http")` should only cover this line.

Reply:
resolved, please review again, thanks.